pkg/apis/kubermatic/v1/helper/helper.go

```diff
@@ -353,3 +354,21 @@ func UpdateResourceQuotaStatus(ctx context.Context, client ctrlruntimeclient.Cli
 		return client.Status().Patch(ctx, resourceQuota, ctrlruntimeclient.MergeFrom(original))
 	})
 }
+
+func HasClusters(ctx context.Context, client ctrlruntimeclient.Client, projectID string) (bool, error) {
+	clusterList := &kubermaticv1.ClusterList{}
+	if err := client.List(ctx, clusterList, ctrlruntimeclient.MatchingLabels{kubermaticv1.ProjectIDLabelKey: projectID}); err != nil {
```

Comment:
You should be able to limit this list to max. 1 item, because fetching more than 1 is wasted effort.